unitary/examples/quantum_chinese_chess/board_test.py

```diff
@@ -155,3 +163,32 @@ def test_flying_general_check():
     board.board["e3"].reset()
     board.board["e6"].reset()
     assert board.flying_general_check() == True
+
+
+def test_flying_general_check_quantum_cases():
+    # When there are quantum pieces in between.
+    board = set_board(["a3", "a4", "e0", "e9"])
+    board.king_locations = ["e0", "e9"]
+    board.current_player = 0  # i.e. RED
+    world = board.board
+    alpha.PhasedSplit()(world["a3"], world["c3"], world["e3"])
+    world["e3"].is_entangled = True
+    alpha.PhasedSplit()(world["a4"], world["c4"], world["e4"])
+    world["e4"].is_entangled = True
+
+    result = board.flying_general_check()
+    # We check the ancilla to learn whether the general/king flies or not.
+    captured = world.post_selection[world["ancilla_ancilla_flying_general_check_0_0"]]
+    if captured:
```

Comment:
It would be nice to know that each branch of the if statement happens. (That some of the time it is captured and some of the time it is not).

One way to get one branch deterministically would be to test this:
```
    board = set_board(["a3", "e0", "e9"])
    board.king_locations = ["e0", "e9"]
    board.current_player = 0  # i.e. RED
    alpha.PhasedSplit()(world["a3"], world["e3"], world["e5"])

```

Reply:
Thanks Doug. I just added this case as a separate test, where capture could not happen. But I cannot think of a way to deterministically make sure the capture happens.